src/legacy/core_plugins/metric_vis/public/metric_vis_fn.js

```diff
@@ -32,21 +32,114 @@ export const metric = () => ({
     defaultMessage: 'Metric visualization'
   }),
   args: {
-    visConfig: {
-      types: ['string', 'null'],
-      default: '"{}"',
+    percentage: {
+      types: ['boolean'],
+      default: false,
+      help: i18n.translate('metricVis.function.percentage.help', {
+        defaultMessage: 'Shows metric in percentage mode. Dont forget to set colorRange.'
+      })
+    },
+    colorSchema: {
+      types: ['string'],
+      default: '"Green to Red"',
+      help: i18n.translate('metricVis.function.colorSchema.help', {
+        defaultMessage: 'Color schema to use'
+      })
+    },
+    colorMode: {
+      types: ['string'],
+      default: '"None"',
+      options: ['None', 'Label', 'Background'],
+      help: i18n.translate('metricVis.function.colorMode.help', {
+        defaultMessage: 'Which part of metric to color'
+      })
+    },
+    colorRange: {
+      types: ['string'],
+      default: '"[{ from: 0, to: 10000 }]"',
+      help: i18n.translate('metricVis.function.colorRange.help', {
+        defaultMessage: 'Color ranges'
```

Comment:
I think more specific help text would be useful here (TBH i'm not sure how colorRange even works)

Reply:
well its just array of objects with from and to property .... we probably want to create another helper function here to generate this objects, and maybe use a multi agg then ? @w33ble  ?

Reply:
Yeah, making users type JSON is pretty much always the wrong thing to do. You could split the args into "from" and "to", or make a new function. I also don't know what this does so it's hard for me to decide which, or offer much more input.

Reply:
...I wonder if it makes sense to put `colorSchema` (or colorScheme), `colorMode`, and `colorRange` into its own function too. Are these concepts shared with other visualizations? If so, even partially, a new function (and type) probably makes sense.